diff --git a/src/common/blkdev.cc b/src/common/blkdev.cc
index 0260a69e470..6e64bc2fcce 100644
--- a/src/common/blkdev.cc
+++ b/src/common/blkdev.cc
@@ -1290,7 +1290,7 @@ void get_device_metadata(
       }
       devpaths += dev + "=" + path;
     } else {
-      (*errs)[dev] + " no unique device path for "s + dev + ": " + err;
+      (*errs)[dev] += " no unique device path for "s + dev + ": " + err;
     }
   }
 }
