diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index f0128e3a3..c07bba20e 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -621,6 +621,7 @@ add_subdirectory(osdc)
 add_subdirectory(perfglue)
 
 add_library(rados_snap_set_diff_obj OBJECT librados/snap_set_diff.cc)
+add_dependencies(rados_snap_set_diff_obj legacy-option-headers)
 
 option(WITH_LIBRADOSSTRIPER "build with libradosstriper support" ON)
 
@@ -881,6 +882,7 @@ if(WITH_RBD)
     add_library(krbd STATIC krbd.cc
       $<TARGET_OBJECTS:parse_secret_objs>)
     target_link_libraries(krbd keyutils::keyutils)
+    add_dependencies(krbd legacy-option-headers)
   endif()
   add_subdirectory(librbd)
   if(WITH_FUSE)
