diff --git a/src/cephadm/build.py b/src/cephadm/build.py
index 53d3983de..7cd78e9cd 100755
--- a/src/cephadm/build.py
+++ b/src/cephadm/build.py
@@ -162,7 +162,7 @@ class Config:
     def __init__(self, cli_args):
         self.cli_args = cli_args
         self._maj_min = sys.version_info[0:2]
-        self.install_dependencies = True
+        self.install_dependencies = False
         self.deps_mode = DependencyMode[cli_args.bundled_dependencies]
         if self.deps_mode == DependencyMode.none:
             self.install_dependencies = False
