--- ceph-19.2.0/src/s3select/include/s3select_parquet_intrf.h.orig	2024-11-26 15:08:27.460400263 -0500
+++ ceph-19.2.0/src/s3select/include/s3select_parquet_intrf.h	2024-11-26 15:09:22.636480526 -0500
@@ -1245,8 +1245,8 @@
   }
 
   file_metadata_ =
-	FileMetaData::Make(metadata_buffer->data(), &metadata_len, file_decryptor_);
-      	//FileMetaData::Make(metadata_buffer->data(), &metadata_len, default_reader_properties(), file_decryptor_); //version>9
+	//FileMetaData::Make(metadata_buffer->data(), &metadata_len, file_decryptor_);
+      	FileMetaData::Make(metadata_buffer->data(), &metadata_len, default_reader_properties(), file_decryptor_); //version>9
 }
 
 void SerializedFile::ParseMetaDataOfEncryptedFileWithPlaintextFooter(
