diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index c07bba20e..83dbe0421 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -905,8 +905,8 @@ endif(WITH_KVS)
 if(WITH_RADOSGW)
   if(WITH_RADOSGW_SELECT_PARQUET OR WITH_RADOSGW_ARROW_FLIGHT)
     if(WITH_SYSTEM_ARROW)
-      find_package(Arrow 4 REQUIRED QUIET)
-      find_package(Parquet 4 REQUIRED QUIET)
+      find_package(Arrow REQUIRED QUIET)
+      find_package(Parquet REQUIRED QUIET)
     else()
       # find arrow's dependencies
       if (WITH_SYSTEM_UTF8PROC)
