Description: Pass build flags when building Python extensions
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/ceph/+bug/2078244
Bug-Gentoo: https://bugs.gentoo.org/866159
Bug-Gentoo: https://bugs.gentoo.org/866161
Origin: gentoo, https://github.com/gentoo/gentoo/commit/c588d0c55a2facb7b98331abf3ccb2439cc34286
Author: Patrick McLean <chutzpah@gentoo.org>
Last-Update: 2022-08-24

--- a/cmake/modules/Distutils.cmake
+++ b/cmake/modules/Distutils.cmake
@@ -93,12 +93,12 @@
     COMMAND
     env
     CC="${PY_CC}"
-    CFLAGS="${PY_CFLAGS}"
+    CFLAGS="${PY_CFLAGS}"\ ${CMAKE_C_FLAGS}
     CPPFLAGS="${PY_CPPFLAGS}"
     CXX="${PY_CXX}"
     LDSHARED="${PY_LDSHARED}"
     OPT=\"-DNDEBUG -g -fwrapv -O2 -w\"
-    LDFLAGS=-L${CMAKE_LIBRARY_OUTPUT_DIRECTORY}
+    LDFLAGS=-L${CMAKE_LIBRARY_OUTPUT_DIRECTORY}\ ${CMAKE_LINKER_FLAGS}\ ${CMAKE_SHARED_LINKER_FLAGS}
     CYTHON_BUILD_DIR=${CMAKE_CURRENT_BINARY_DIR}
     CEPH_LIBDIR=${CMAKE_LIBRARY_OUTPUT_DIRECTORY}
     ${Python3_EXECUTABLE} ${setup_py}
